index.js

```diff
@@ -99,7 +100,7 @@ Request.prototype._tryUntilFail = function () {
       response.attempts = this.attempts;
     }
     if (this.retryStrategy(err, response, body) && this.maxAttempts > 0) {
-      this._timeout = setTimeout(this._tryUntilFail.bind(this), this.retryDelay);
+      this._timeout = setTimeout(this._tryUntilFail.bind(this), this.delayStrategy(err, response, body));
```

Comment:
I was thinking that since we are calling delayStrategy from this, the context does get preserved. I went through the debugger to double check that it was and it checked out. However, it doesn't hurt to be more explicit.